test/extended/apiserver/apply.go

```diff
@@ -0,0 +1,168 @@
+package apiserver
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	discocache "k8s.io/client-go/discovery/cached"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	exetcd "github.com/openshift/origin/test/extended/etcd"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+// TODO(marun) Evaluate the CRDs that ocp with (~70 at time of
+// writing) which will involve creating the associated stubs.
+var _ = g.Describe("[sig-api-machinery][Feature:ServerSideApply] Server-Side Apply", func() {
+	var (
+		// mapper is used to translate the gvr provided by etcd
+		// storage data to the gvk required to create correct resource
+		// yaml.
+		mapper *restmapper.DeferredDiscoveryRESTMapper
+	)
+
+	defer g.GinkgoRecover()
+
+	// Defer project creation to tests that require it by calling
+	// NewCLIWithoutNamespace instead of NewCLI.
+	oc := exutil.NewCLIWithoutNamespace("server-side-apply")
+
+	g.BeforeEach(func() {
+		// Only init once per worker
+		if mapper != nil {
+			return
+		}
+		kubeClient, err := kubernetes.NewForConfig(oc.AdminConfig())
+		o.Expect(err).NotTo(o.HaveOccurred())
+		mapper = restmapper.NewDeferredDiscoveryRESTMapper(discocache.NewMemCacheClient(kubeClient.Discovery()))
```

Comment:
nit: why a new one for each subtest?

Reply:
Note the conditional check above. It's initialized once per worker, not once per subtest. My ginkgo-foo is weak, if there's a better way to accomplish this I'm all ears!